15.md

```diff
@@ -47,7 +47,6 @@ A merchant can publish these events:
 
 ```json
 {
-  "id": <string, id generated by the merchant. Sequential IDs (`0`, `1`, `2`...) are discouraged>,
   "name": <string, stall name>,
   "description": <string (optional), stall description>,
   "currency": <string, currency used>,
```

Comment:
Why is there a single currency for the stall? The merchant should be able to specify multiple currencies that they are willing to accept.

Reply:
The `currency` can be `BTC` or a fiat `currency`.
The payment only takes place using BTC.  **There is no payment in fiat**.

The fiat currency serves as an indicator, it will be converted to `sats` at the moment the invoice is created.

Having the merchant set **multiple currencies** (lets say Turkish Lira and USD)would:
 - require the merchant to keep track of the exchange rate between these currencies and update the prices accordingly
 - allow the clients to select the fiat currency that has the best exchange rate against BTC, thus paying less

However, on the client side (not merchant side), the application can take one fiat currency (lets say USD) and show the user the value in any currency it requires (eg: Turkish Lira)

Reply:
An indicator of what?

How would the value of this currency be determined without the merchant being required to keep track of it?

Reply:
I don't understand how this is even a question. You have one currency that the seller is willing to accept: sats and one fiat currency that the seller is willing to have the price fixed in. The clients can display anything else, but that is a UI issue, not a protocol issue. You can't have more than one fiat currency here, because fiat currencies fluctuate against each other, so it would be impossible to make the conversion to sats at the time of sale!

Reply:
> Why is there a single currency for the stall? The merchant should be able to specify multiple currencies that they are willing to accept.

The merchant does not only specify the currency, it also has to specify the actual value.
Eg for today:
 - 10 USD
 - 300 Turkish Lira
 
 What will be the prices tomorrow if the Turkish Lira depreciates by 25%?

Reply:
> I don't understand how this is even a question. You have one currency that the seller is willing to accept: sats and one fiat currency that the seller is willing to have the price fixed in. The clients can display anything else, but that is a UI issue, not a protocol issue. You can't have more than one fiat currency here, because fiat currencies fluctuate against each other, so it would be impossible to make the conversion to sats at the time of sale!

Thank you, it makes more sense now.  If I understand correctly, "currency" is an indicator of BTC value in the merchant's preferred fiat currency.  Basically a, "merchant's price display preference", yes?

Reply:
> > Basically a, "merchant's price display preference", yes?
> 
> Not really. It is the amount of fiat currency that the merchant is willing to sell the item for. As a merchant you can say you want to sell the T-shirt for 20 USD or for 30k IDR... and it will always sell for that amount regardless of the USD <=> IDR conversion. This is not about price display. It is about _price_. Price display can be anything, up to the UI implementation.

Is there a portion of this NIP that explains how the BTC amount is updated to stay at the fiat value (which I'm assuming would be in the Product definition); or is this something each marketplace client will determine on their own?

Reply:
> Is there a portion of this NIP that explains how the BTC amount is updated to stay at the fiat value

No, it is not necessary for that to be part of the protocol.

Please experiment with one of the implementations (plebian-market or the LNbits nostrmarket extension), it will make things clearer.

Reply:
The BTC amount is not updated at any point... it is simply the fiat amount converted to BTC at the time the buyer tries to buy something.

Reply:
The "converted" part is what I'm asking about.  I don't see that mentioned in this NIP.  Is there an assumed mechanism?

Reply:
NIPs define the communication protocol between clients. Currency conversions are not part of the protocol. The seller could be manually opening coinbase.com and looking up the price. It really does not matter.

Reply:
Yes, as others have commented, prices are fixed in a currency and converted to the market value of BTC at the moment when a user wants to buy the product. I also think that from a protocol perspective, there is no need to specify how to convert between currency/BTC, this is something specific and the responsibility of the platform